app/questionnaire_state/state_answer.py

```diff
@@ -38,6 +41,16 @@ def _restore_other_value(self, user_input):
                     self.value = option['value']
                     self.other = self.schema_item.get_user_input(user_input)
 
+                elif self.input == other_option['value'] and hasattr(user_input, 'getlist'):
```

Comment:
Yeah, unfortunately the logic gets called on GET and POST. When POST-ing an `ImmutableMultiDict` gets passed in. This whole `other` solution needs to be re-visited, I feel.
At the moment we're sort of storing an answer inside an answer and that causes quite a lot of unnecessary complexity - perhaps we can change this when we bring in the WTF-Forms solution.